File: main workshop/services/dotnet/ride-completion-service/Models/PaymentCompletedEvent.cs
using System.Text.Json.Serialization;

/// <summary>
/// Represents a payment completion event received from EventBridge
/// </summary>
public class PaymentCompletedEvent
{
    /// <summary>
    /// Unique identifier for the payment
    /// </summary>
    [JsonPropertyName("paymentId")]
    public string PaymentId { get; set; } = string.Empty;

    /// <summary>
    /// Unique identifier for the ride associated with the payment
    /// </summary>
    [JsonPropertyName("rideId")]
    public string RideId { get; set; } = string.Empty;

    /// <summary>
    /// Unique identifier for the rider who made the payment
    /// </summary>
    [JsonPropertyName("riderId")]
    public string RiderId { get; set; } = string.Empty;

    /// <summary>
    /// Unique identifier for the driver who completed the ride
    /// </summary>
    [JsonPropertyName("driverId")]
    public string DriverId { get; set; } = string.Empty;

    /// <summary>
    /// Payment amount
    /// </summary>
    [JsonPropertyName("amount")]
    public decimal Amount { get; set; }

    /// <summary>
    /// Payment method used (e.g., credit-card, somecompany-pay, etc.)
    /// </summary>
    [JsonPropertyName("paymentMethod")]
    public string PaymentMethod { get; set; } = string.Empty;

    /// <summary>
    /// Unique identifier for the payment transaction
    /// </summary>
    [JsonPropertyName("transactionId")]
    public string TransactionId { get; set; } = string.Empty;

    /// <summary>
    /// ISO8601 timestamp when the payment was completed
    /// </summary>
    [JsonPropertyName("timestamp")]
    public string Timestamp { get; set; } = string.Empty;

    /// <summary>
    /// Correlation ID for tracking requests across services
    /// </summary>
    [JsonPropertyName("correlationId")]
    public string? CorrelationId { get; set; }
}

File: main workshop/services/java/ride-completion-service/src/main/java/com/powertoolsride/ridecompletionservice/model/PaymentCompletedEvent.java
package com.powertoolsride.ridecompletionservice.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.math.BigDecimal;

/**
 * Represents a payment completion event received from EventBridge
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record PaymentCompletedEvent(
    @JsonProperty("paymentId") String paymentId,
    @JsonProperty("rideId") String rideId,
    @JsonProperty("riderId") String riderId,
    @JsonProperty("driverId") String driverId,
    @JsonProperty("amount") BigDecimal amount,
    @JsonProperty("paymentMethod") String paymentMethod,
    @JsonProperty("transactionId") String transactionId,
    @JsonProperty("timestamp") String timestamp,
    @JsonProperty("correlationId") String correlationId
) {}
